From 5524cf06631e777629e62ec6aa998148476a35b6 Mon Sep 17 00:00:00 2001
From: Rudi Heitbaum <rudi@heitbaum.com>
Date: Tue, 30 Dec 2025 15:34:17 +1100
Subject: [PATCH] Add climits includes

Fixes:
../libheif/api/libheif/heif_image_handle.cc: In function 'int heif_image_handle_get_width(const heif_image_handle*)':
../libheif/api/libheif/heif_image_handle.cc:48:13: error: 'INT_MAX' was not declared in this scope
   48 |     if (w > INT_MAX) {
      |             ^~~~~~~
../libheif/api/libheif/heif_image_handle.cc:23:1: note: 'INT_MAX' is defined in header '<climits>'; this is probably fixable by adding '#include <climits>'
   22 | #include "api_structs.h"
  +++ |+#include <climits>
   23 | #include <string>
---
 libheif/api/libheif/heif_image_handle.cc | 1 +
 1 file changed, 1 insertion(+)

diff --git a/libheif/api/libheif/heif_image_handle.cc b/libheif/api/libheif/heif_image_handle.cc
index ada292712f..9080ce4a52 100644
--- a/libheif/api/libheif/heif_image_handle.cc
+++ b/libheif/api/libheif/heif_image_handle.cc
@@ -20,6 +20,7 @@
 
 #include "heif_image_handle.h"
 #include "api_structs.h"
+#include <climits>
 #include <string>
 
 
